Trying to implement without CordovaActivity

I am trying to put this to work without extending CordovaActivity. I need to do this because I am using another library for which I need to extend FragmentActivity.

At the moment I am getting the following errors:

```
MainActivity.java:108: error: cordovaPluginAdapter has protected access in CordovaPluginPackage
        CordovaInterfaceImpl cordovaInterface = cordovaPluginPackage.cordovaPluginAdapter.cordovaInterface;
                                                                    ^
MainActivity.java:108: error: cordovaInterface has protected access in CordovaPluginAdapter
        CordovaInterfaceImpl cordovaInterface = cordovaPluginPackage.cordovaPluginAdapter.cordovaInterface;
```

My code:

```
package com.realityConnect.mckrides;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;

import com.facebook.react.LifecycleState;
import com.facebook.react.ReactInstanceManager;
import com.facebook.react.ReactRootView;
import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;

import android.support.v4.app.FragmentActivity;
import com.aakashns.reactnativedialogs.ReactNativeDialogsPackage;

import io.cordova.reactnative.CordovaPluginPackage;
import org.apache.cordova.CordovaInterfaceImpl;
import org.apache.cordova.LOG;

public class MainActivity extends FragmentActivity implements DefaultHardwareBackBtnHandler {

    private ReactInstanceManager mReactInstanceManager;
    private ReactRootView mReactRootView;
    private static final String TAG = "CORDOVA_ACTIVITY";
    protected CordovaPluginPackage cordovaPluginPackage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mReactRootView = new ReactRootView(this);

        mReactInstanceManager = ReactInstanceManager.builder()
                .setApplication(getApplication())
                .setBundleAssetName("index.android.bundle")
                .setJSMainModuleName("index.android")
                .addPackage(new MainReactPackage())

                .addPackage(new CordovaPluginPackage(this, savedInstanceState))
                .addPackage(new ReactNativeDialogsPackage(this))

                .setUseDeveloperSupport(true)
                .setInitialLifecycleState(LifecycleState.RESUMED)
                .build();

        mReactRootView.startReactApplication(mReactInstanceManager, "McRides", null);

        setContentView(mReactRootView);
    }

...

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
        CordovaInterfaceImpl cordovaInterface = cordovaPluginPackage.cordovaPluginAdapter.cordovaInterface;
        LOG.d(TAG, "Incoming Result. Request code = " + requestCode);
        super.onActivityResult(requestCode, resultCode, intent);
        cordovaInterface.onActivityResult(requestCode, resultCode, intent);
    }
}

```

--- a/framework/android/src/main/java/io/cordova/reactnative/CordovaPluginAdapter.java
+++ b/framework/android/src/main/java/io/cordova/reactnative/CordovaPluginAdapter.java
@@ -24,7 +24,7 @@ public class CordovaPluginAdapter extends ReactContextBaseJavaModule {
     private static final String TAG = "CordovaPluginAdapter";
     private CordovaPreferences preferences;
     private ArrayList<PluginEntry> pluginEntries;
-    protected CordovaInterfaceImpl cordovaInterface;
+    public CordovaInterfaceImpl cordovaInterface;
 
     public CordovaPluginAdapter(ReactApplicationContext reactContext, Activity mainReactActivity, Bundle savedInstanceState) {
         super(reactContext);

--- a/framework/android/src/main/java/io/cordova/reactnative/CordovaPluginPackage.java
+++ b/framework/android/src/main/java/io/cordova/reactnative/CordovaPluginPackage.java
@@ -16,7 +16,7 @@ import java.util.List;
 public class CordovaPluginPackage implements ReactPackage {
     protected final Bundle savedInstanceState;
     protected Activity activity;
-    protected CordovaPluginAdapter cordovaPluginAdapter;
+    public CordovaPluginAdapter cordovaPluginAdapter;
 
     public CordovaPluginPackage(Activity reactActivity, Bundle bundle) {
         savedInstanceState = bundle;

